src/NuGet.Core/NuGet.Protocol/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -56,6 +58,11 @@ private HttpHandlerResourceV3 CreateResource(PackageSource packageSource)
                 AutomaticDecompression = (DecompressionMethods.GZip | DecompressionMethods.Deflate),
             };
 
+            if (packageSource.DisableTLSCertificateValidation)
+            {
+                clientHandler.ServerCertificateCustomValidationCallback = (HttpRequestMessage message, X509Certificate2 cert, X509Chain chain, SslPolicyErrors errors) => true;
```

Comment:
This feels broader than just disabling TLS validation. Is it safe?

Perhaps at least check the enum. If I understand the test:
`=> errors == SslPolicyErrors.RemoteCertificateChainErrors`

Reply:
I see that there's [a proposal ](https://github.com/NuGet/Home/pull/13024)that calls this out and dismisses the concern about it being a global disabling of all certificate errors.
However, I don't see that PR approved by @dtivel, and I'd be very interested to hear his perspective on it.

Reply:
That is a good point. However, I am not sure what you mean by globally. If you meant, disabling TLS globally across all sources, that would be a huge security risk definitely. I think the goal is to disable TLS certification completely, however, per source level.  Please let me know if the code doesn't do that. I have updated the doc to mention that this is done per source level.

Reply:
Earlier iterations of @JonDouglas's spec did have a global option for disabling TLS certificate verification for all package sources.  Based on feedback (like [this](https://github.com/NuGet/Home/pull/12542#discussion_r1178157422)), the proposal for a global option was [moved](https://github.com/NuGet/Home/pull/12542/commits/ac74ed18f1e82a5cdc9d9604fa2341ca702e14f2) to a ["redacted" section](https://github.com/NuGet/Home/blob/dev/proposed/2023/InsecureConnectionsDisableCertificateValidation.md#redacted-functional-experiences-based-on-community-feedback), meaning you can still see the rejected global option proposal.  The final version of the spec does not support an option for global disablement.  You could have the same net effect by disabling verification on every package source, but a single global option does not exist.

As far as [this feedback](https://github.com/NuGet/NuGet.Client/pull/5514/files#r1397951051) goes, the spec was clear that all verification should be disable-able on a per package source basis.  While the verification _could_ have had more granular knobs (e.g.:  respect some verification failures but ignore others), the spec landed (with community approval) on ignoring all verification failures with one setting on a per package source basis.

+1 to @kartheekp-ms's [excellent suggestion](https://github.com/NuGet/NuGet.Client/pull/5514/files#r1398023860).